agent/ec2/ec2_client.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package ec2
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/ecs_client/model/ecs"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	ec2sdk "github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/cihub/seelog"
+)
+
+const (
+	clientRetriesNum = 5
+
+	// InstanceMaxTagsNum is the maximum number of tags that an instance can have.
+	InstanceMaxTagsNum = 50
```

Comment:
I wouldn't hardcode this expectation -- just in case they change in a future version.

Reply:
Well I don't want to hard code it as well, the API provides this parameter but it doesn't mention the default value, this number can be 5-1000, so we end up 3 options:
1. Use 50, as it's mentioned in EC2 public doc that an instance can have a maximum of 50 tags.
2. Use 1000, and assume that the max num of tags won't be changed to a number greater than 1000.
3. Don't provide this parameter or use a number like 50 (or even 1000), make multiple calls until we cannot get more tags.

Which one do you prefer? I'd say option 3 with the max tags number being 1000 is the safest option, looks like a bit wired tho as we already know 50 would be enough.

Reply:
Do we know if customers will ever have >50 tags?

Reply:
Yes, docs will test customer that the max num of tags for each ECS resource is 50.

Reply:
Makes sense!